### PR TITLE
[WIP][OCPCLOUD-1381] Implement AWS Placement Group support and creation

### PR DESCRIPTION
**Pull request by JoelSpeed**
This PR implements the changes required to allow users to add Machines to placement groups.

Additionally, should the placement group not exist, this PR allows the provider to create the placement group based on some pre-existing configuration within the placement struct.

This is currently blocked on https://github.com/openshift/api/pull/1091, hence being marked WIP

### CODE REVIEW

File: pkg/actuators/machine/instances.go [hunk 1]
```diff
@@ -505,3 +488,107 @@ func getInstanceMarketOptionsRequest(providerConfig *machinev1.AWSMachineProvide
...
+	}
+
+	return nil
+}
+
+// isAWS4xxError will determine if the passed error is an AWS error with a 4xx status code.
+func isAWS4xxError(err error) bool {
+	if _, ok := err.(awserr.Error); ok {
+		if reqErr, ok := err.(awserr.RequestFailure); ok {
+			if strings.HasPrefix(strconv.Itoa(reqErr.StatusCode()), "4") {
```
Comment (elmiko) [inline pkg/actuators/machine/instances.go:588]: do we need to convert the int to ascii here, couldn't we just check `StatusCode() >= 400 && StatusCode() < 500`?

**Review by elmiko — COMMENTED** [review]
this generally makes sense to me, i just have one minor question

File: pkg/actuators/machine/instances.go (hunk 1 above)
Comment (JoelSpeed) [reply pkg/actuators/machine/instances.go:588]: This was just copied from https://github.com/openshift/machine-api-provider-aws/pull/16/files#diff-c5079a7c5d2916c3661d0c44d1267050b3496c40b4530363d3825d66af581b6fR376, but you're right, that would make more sense, I'll update it

Comment (elmiko) [reply pkg/actuators/machine/instances.go]: no worries, i just thought converting to ascii seemed like a needless use of cycles

**Review by elmiko — COMMENTED** [review]
thanks for the update, i'm good with this.
/approve

**Comment by JoelSpeed** [comment]
This was superseded by other work

**Pull request closed without merging**